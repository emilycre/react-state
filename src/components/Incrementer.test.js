import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('Incrementer component', () => {
  it('Renders an incrementer', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('Increments a count on button click', () => {
    const wrapper = shallow(<Incrementer />);

    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('count')).toEqual(1);
  });
});
