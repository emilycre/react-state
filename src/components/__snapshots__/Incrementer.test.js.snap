// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incrementer component Incremembers a count on button click 1`] = `
<Fragment>
  <p>
    Current count: 
    1
  </p>
  <button
    onClick={[Function]}
  >
    Increment.
  </button>
</Fragment>
`;

exports[`Incrementer component Renders an incrementer 1`] = `
<Fragment>
  <p>
    Current count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment.
  </button>
</Fragment>
`;
